### ISSUE TITLE
Invalid patterns are silently discarded when constructing a SchemaDocument

### ISSUE THREAD

**Issue by trink** [miloyip/rapidjson#537]
```json
"numeric_string":{            
"type":"string",
"pattern":"^\\d+$"
}
```
CreatePattern fails with an invalid (internal regex) and the pattern test is silently removed from the schema check.

**Comment by miloyip** [comment]
Yes. Currently schema keyword with invalid or unsupported values are silently ignored.

I haven't thought of this problem before because I think we can simply validate a JSON schema with the JSON meta-schema. However, the contents of regex cannot be validated by JSON meta-schema.

Do you have any idea on how to handle the unsupported regex situation?
